Feature/spinner wheel

lejerin opened the pull request:
룰렛 기본 구현

이슈 : 룰렛을 왼쪽으로 swipe해도 오른쪽으로 돌아감 (의견 환영)

eunjanii committed:
돌림판 기본 구현

RouletteView 커스텀뷰

- 내부 원 그리기 완료
- 테두리, 구분선 그리기 완료
- 라벨 그리기 완료

eunjanii committed:
돌림판 회전 구현

eunjanii committed:
merge develop

eunjanii committed:
룰렛 결과값 가져오기

view가 270도 지점부터 그려지므로 90도를 더한 뒤 360의 나머지를 구해서 인덱스를 계산

eunjanii committed:
룰렛 디자인 변경

eunjanii committed:
텍스트 라운드 처리

eunjanii committed:
룰렛 Swipe 제스쳐 추가 & Bottom Margin 수정

Swipe x 축 길이에 따라 회전 각도 계산하여 회전

Todo: x축이 아니라 경로 길이를 계산 & 속도 비례하여 회전 계산

eunjanii committed:
Fragment bundle 데이터 초기화 및 Fling 이벤트 수정

이슈 : 왼쪽으로 스와이프해도 룰렛이 돌아감

eunjanii committed:
Merge branch 'develop' into feature/spinner_wheel

eunjanii committed:
Merge branch 'develop' into feature/spinner_wheel

eunjanii committed:
merge Develop

eunjanii committed:
fixed untracked files

wnehdals commented on app/src/main/java/com/jdm/menunenaw/ui/roulette/RouletteFragment.kt:
```diff
@@ -0,0 +1,62 @@
+        rouletteView.setRouletteData(
+            arguments?.getStringArrayList(BUNDLE_DATA_ROULETTE_LIST)?.toList() ?: listOf("데이터는", "최소", "3개")
+        )
+        rouletteView.setOnRouletteResultListener(mutableLiveData) {
```
mutablelivedata를 넘기는것말고 setRouletteData 처럼 setListner함수만들어서 리스너객체로 결과값을 가져오는것이 더 좋아보입니다.
리스너 객체에 파라미터로 함수를 넘겨 결과값을 받게하면 리스너에서도 라이브데이터를 안써도 될 것 같습니다.

eunjanii committed:
의견 반영

1. BundleKey 정의 : BaseValue BundleKey enum class 내부에 정의
2. RouletteView 인터페이스 리스너로 변경 -> Fragment 에서는 viewModel 로 데이터 전달하여, LiveData postValue() 호출
3. 룰렛 내부 구분선 그리는 방식 변경
